var/spack/repos/builtin/packages/hipace/package.py

```diff
@@ -43,9 +43,14 @@ class Hipace(CMakePackage):
         depends_on('openpmd-api ~mpi', when='~mpi')
         depends_on('openpmd-api +mpi', when='+mpi')
     with when('compute=omp'):
-        depends_on('fftw@3: +openmp')
-        depends_on('fftw ~mpi', when='~mpi')
-        depends_on('fftw +mpi', when='+mpi')
+        # FFTW: general API, implemented by various packages
+        depends_on('fftw-api@3 +openmp')
+        depends_on('fftw-api ~mpi', when='~mpi')
+        depends_on('fftw-api +mpi', when='+mpi')
```

Comment:
We have no variants on virtual packages atm, but we can enforce a behavior in case a provider is selected as virtual, see https://spack.github.io/spack-cscs2024/#/9/3

That said, is this PR something you'd still want?